docs/guides/basic-rbac.mdx

```diff
@@ -3,267 +3,303 @@ title: Implement basic Role Based Access Control (RBAC) with metadata
+  - `unsafeMetadata` - can be both read and modified in the browser, meaning users can modify this data. Therefore, you should validate `unsafeMetadata` before trusting it for sensitive operations.
+  - `privateMetadata` - cannot be accessed nor modified in the browser.
+  - `publicMetadata` - can be accessed in the browser, but only be modified server-side or in the Clerk Dashboard.
 
-In the Clerk Dashboard, navigate to [**Sessions**](https://dashboard.clerk.com/last-active?path=sessions). In the **Customize session token** section, select **Edit**. In the modal that opens, enter the following JSON. If you have already customized your session token, you may need to merge this with what you currently have.
+  Since `publicMetadata` can only be read but not modified in the browser, it is the safest and most appropriate choice for storing information
 
-```json
-{
-  "metadata": "{{user.public_metadata}}"
-}
-```
+  To build a basic RBAC system, you first need to make `publicMetadata` available to the application directly from the session token. By attaching `publicMetadata` to the user's session, you can access the data without needing to make a fetch or network request each time.
 
-![The Sessions page in the Clerk Dashboard with the 'Customize session token' modal opened. The modal has a text field with the JSON 'metadata': '\{\{user.public\_metadata}}'.](/docs/images/guides/basic-rbac/customize-session-token.webp)
+  1. Navigate to the [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=sessions).
+  1. In the top navigation, select **Configure**. Then in the sidebar, select **Sessions**.
+  1. Under the **Customize session token** section, select **Edit**.
+  1. In the modal that opens, enter the following JSON and select **Save**. If you have already customized your session token, you may need to merge this with what you currently have.
 
-> [!CAUTION]
-> The entire session token has a limit of 4kb of data. Exceeding this size can have adverse effects, including a possible infinite redirect loop for users who exceed this size in Next.js applications.
-> It's recommended to move particularly large claims out of the JWT and fetch these using a separate API call from your backend.
+  ```json
+  {
+    "metadata": "{{user.public_metadata}}"
+  }
+  ```
+
+  > [!CAUTION]
+  > The session token has a 4KB size limit. Exceeding this limit can have adverse effects, such as an infinite redirect loop for users in Next.js applications.
+  > To avoid this, it's recommended to move large claims out of the JWT and fetch them via a separate API call from your backend.
+
+  ### Create a global TypeScript definition
+
+  1. In your application's root folder, create a `types` directory.
+  1. Inside this directory, add a `globals.d.ts` file. This file will provide auto-completion, prevent TypeScript errors when working with roles, and control which roles are allowed in the application.
```

Comment:
```suggestion
  1. Inside this directory, add a `globals.d.ts` file. This file will provide auto-completion and prevent TypeScript errors when working with roles.
```

> and control which roles are allowed in the application
This isn't really true. If you are using TypeScript and respecting linting then the app won't deploy with the errors but that's less about controlling the roles than alerting you to errors where you specific another role. Also the server action that sets the metadata/role doesn't check its one of the roles in `Roles` cause its a super basic example.